service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/AllocationDecisionController.java

```diff
@@ -48,6 +48,19 @@ public AboutToStartOrSubmitCallbackResponse checkIfAllocationProposalIsMissing(
 
         decisionBuilder.allocationProposalPresent(checkIfAllocationProposalIsPresent(caseData.getAllocationProposal()));
 
+        if (checkIfAllocationProposalIsPresent(caseData.getAllocationProposal()).equals("Yes")) {
```

Comment:
nested ifs make me want to vomit 🤢 

there must be a better way to do this 🤔 I will have a think